packages/overlay/test/interactions.test.js

```diff
@@ -0,0 +1,298 @@
+import { expect } from '@esm-bundle/chai';
+import {
+  click,
+  enterKeyDown,
+  escKeyDown,
+  fixtureSync,
+  mousedown,
+  mouseup,
+  nextRender,
+  oneEvent,
+} from '@vaadin/testing-helpers';
+import sinon from 'sinon';
+import '../vaadin-overlay.js';
+import { createOverlay } from './helpers.js';
+
+describe('interactions', () => {
+  describe('Esc', () => {
+    let overlay;
+
+    beforeEach(async () => {
+      overlay = createOverlay('overlay content');
+      overlay.opened = true;
+      await oneEvent(overlay, 'vaadin-overlay-open');
+    });
+
+    afterEach(async () => {
+      overlay.opened = false;
+      await nextRender();
+    });
+
+    it('should close on Esc', () => {
+      escKeyDown(document.body);
+
+      expect(overlay.opened).to.be.false;
+    });
+
+    it('should fire the vaadin-overlay-escape-press event on Esc', () => {
+      const spy = sinon.spy();
+      overlay.addEventListener('vaadin-overlay-escape-press', spy);
+
+      escKeyDown(document.body);
+
+      expect(spy.calledOnce).to.be.true;
+    });
+
+    it('should not fire the vaadin-overlay-escape-press event on other key press', () => {
+      const spy = sinon.spy();
+      overlay.addEventListener('vaadin-overlay-escape-press', spy);
+
+      enterKeyDown(document.body);
+
+      expect(spy.called).to.be.false;
+    });
+
+    it('should not close on esc if vaadin-overlay-escape-press event was prevented', () => {
+      overlay.addEventListener('vaadin-overlay-escape-press', (e) => e.preventDefault());
+
+      escKeyDown(document.body);
+
+      expect(overlay.opened).to.be.true;
+    });
+  });
+
+  describe('click', () => {
+    let parent, overlay, overlayPart, backdrop;
+
+    beforeEach(async () => {
+      parent = document.createElement('div');
+      overlay = fixtureSync('<vaadin-overlay></vaadin-overlay>', parent);
+      overlay.renderer = (root) => {
+        root.textContent = 'overlay content';
+      };
+      await nextRender();
+      overlayPart = overlay.$.overlay;
+      backdrop = overlay.$.backdrop;
+      overlay.opened = true;
+      await oneEvent(overlay, 'vaadin-overlay-open');
+    });
+
+    afterEach(async () => {
+      overlay.opened = false;
+      await nextRender();
+    });
+
+    describe('close on click', () => {
+      it('should not close on inside click', () => {
+        click(overlayPart);
+
+        expect(overlay.opened).to.be.true;
+      });
+
+      it('should close on outside click', () => {
+        click(parent);
+
+        expect(overlay.opened).to.be.false;
+      });
+
+      it('should close on backdrop click', () => {
+        overlay.withBackdrop = true;
+
+        click(backdrop);
+
+        expect(overlay.opened).to.be.false;
+      });
+    });
+
+    describe('vaadin-overlay-outside-click event', () => {
+      it('should fire the vaadin-overlay-outside-click event on outside click', () => {
```

Comment:
**nit:** I guess we could just say `the event` as the event name is already mentioned in the test suite's name.

```suggestion
      it('should fire the event on outside click', () => {
```

Reply:
Thanks, done.